src/store/roadmaps/slice.ts

```diff
@@ -0,0 +1,59 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { Roadmap } from '../../types';
+import { SearchListResponse, searchRoadmaps } from './thunks';
+
+const initialState: RoadmapState = {
+  roadmaps: [],
+  isLoading: false,
+  error: null,
+};
+
+export interface RoadmapState {
+  roadmaps: Roadmap[];
+  isLoading: boolean;
+  error: string | null;
+}
+
+export const roadmapsSlice = createSlice({
+  name: 'roadmaps',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(searchRoadmaps.pending, onPending)
+      .addCase(searchRoadmaps.fulfilled, onSuccess)
+      .addCase(searchRoadmaps.rejected, onRejected);
+  },
+});
+
+const onSuccess = (
+  state: RoadmapState,
+  action: PayloadAction<SearchListResponse>,
+) => {
+  console.log('Received payload:', action.payload);
+  const roadmapsIds = state.roadmaps.map((roadmap) => roadmap.pk);
+  const filteredRoadmaps = action.payload.results.filter(
+    (roadmap) => !roadmapsIds.includes(roadmap.pk),
+  );
+  state.roadmaps = [...state.roadmaps, ...filteredRoadmaps];
+  state.isLoading = false;
+  state.error = null;
+};
+
+const onPending = (state: RoadmapState) => {
+  state.isLoading = true;
+  state.error = null;
+};
+
+const onRejected = (
+  state: RoadmapState,
+  action: PayloadAction<string | undefined>,
```

Comment:
If you have a rejected action, you'll have had a string in rejectWithValue func, and you shouldn't write  <string | undefined>, use ! operator on 54 line instead (action.payload!) - it guaranties that payload is always in reject action